cloud/services/agentpools/agentpools.go

```diff
@@ -30,14 +30,15 @@ import (
 
 // Spec contains properties to create a agent pool.
 type Spec struct {
-	Name          string
-	ResourceGroup string
-	Cluster       string
-	Version       *string
-	SKU           string
-	Replicas      int32
-	OSDiskSizeGB  int32
-	VnetSubnetID  string
+	Name              string
+	ResourceGroup     string
+	Cluster           string
+	Version           *string
+	SKU               string
+	Replicas          int32
+	OSDiskSizeGB      int32
+	VnetSubnetID      string
+	AvailabilityZones *[]string
```

Comment:
do you prefer pointer vs non-pointer?

Reply:
Hmm, just kept it in sync with the azure api. Is there a reason to not pick a pointer here ?